Update introduction of main cell type report (#523)
AlexsLemonade/scpca-nf: templates/qc_report/celltypes_qc.rmd

```

<!-- Background depends on which cell types are present in the SCE. -->

> This section details results from performing cell type annotation:

Reviewer: I'm not 100% sure about keeping this sentence as it is currently worded because we are planning to include this as a child rmd in the supplementary cell type report. But, this is clearly the right sentence to have at the top of this section as it appears in the main report!

> This section details results from performing cell type annotation:

Reviewer: But, make it a period if we do keep it - 
```suggestion
This section details results from performing cell type annotation.
```

> This section details results from performing cell type annotation.

Author: ```suggestion
The plots and tables included here detail the results from performing cell type annotation.
```
How about something more generic like this? I think the key here is removing the mention of "this section" and then it can be used in both reports. That being said, I do think when we address https://github.com/AlexsLemonade/scpca-nf/issues/514, we may be changing the introduction of the supplemental report regardless. So it might make sense to separate out the intro statements here to only have them in the main report and not include in the supplemental. I think that's a decision we should make when we get the supplemental intro done.

> Cell type annotations were obtained in the following way: 

Reviewer: ```suggestion
Cell type annotations were obtained in the following way(s): 
```
